pyproject.toml

```diff
@@ -0,0 +1,173 @@
+    "Jonas Hörsch <coroa@unknown.com>",
+    "Paul Natsuo Kishimoto <kishimot@iiasa.ac.at>",
+    "Thorsten Burandt <tburandt@unknown.com>",
+    "Ross Ursino <rossursino@unknown.com>",
+    "Maik Budzinski <mabudz@unknown.com>",
+    "Jarmo Kikstra <kikstra@iiasa.ac.at>",
+    "Michael Pimmer <fonfon@unknown.com>",
+    "Patrick Jürgens <pjuergens@unknown.com>",
+    "Florian Maczek <maczek@iiasa.ac.at>",
+    "Laura Wienpahl <LauWien@unknown.com>",
+    "Philip Hackstock <hackstock@iiasa.ac.at>",
+    "Pietro Monticone <pitmonticone@unknown.com>",
+    "Edward Byers <byers@iiasa.ac.at>",
+    "Fridolin Glatter <glatter@iiasa.ac.at>",
+]
+license = "Apache-2.0"
+readme = "README.md"
+packages = [{ include = "pyam" }]
+repository = "https://github.com/IAMconsortium/pyam"
+documentation = "https://pyam-iamc.readthedocs.io"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Science/Research",
+    "Topic :: Scientific/Engineering",
+    "Natural Language :: English",
+    "Programming Language :: Python",
+]
+
+# NOTE TO DEVS
+# If you change a minimum version below, please explicitly implement the change
+# in our minimum-reqs test in the file ./.github/workflows/pytest-dependency.yml
+# Please also add a section "Dependency changes" to the release notes
+[tool.poetry.dependencies]
+# httpx[http2] included via ixmp4    
+# requests  included via ixmp4
+python = ">=3.10, <3.13"
+iam-units = ">=2020.4.21"
+ixmp4 = ">=0.7.1"
+matplotlib = ">=3.6.0"
+numpy = ">=1.26.0"
+openpyxl = ">=3.1.2"
+pandas = ">=2.1.2"
+Pint = ">=0.13"
+PyYAML = ">=6.0.1"
+scipy = ">=1.10.0"
+seaborn = ">=0.11"
+six = ">=1.16.0"
+setuptools = ">=69"
+setuptools-scm = ">=8.0.4"
+wquantiles = ">=0.6"
+XlsxWriter = ">=3.2.0"
+
+[tool.poetry.group.dev]
+optional = true
+
+[tool.poetry.group.dev.dependencies]
+ruff = ">=0.2.1"
+
+[tool.poetry.group.docs]
+optional = true
+
+[tool.poetry.group.docs.dependencies]
+# docs requires 'tutorials'
+autodocsumm = ">=0.2.12"
+kaleido = ">=0.2.0, <0.2.1.post"
+nbsphinx = ">=0.9.3"
+numpydoc = ">=1.6.0"
+pillow = ">=10.2.0"
+sphinx = ">=7.2.6"
+sphinx-gallery = ">=0.15.0"
+
+[tool.poetry.group.optional_io_formats]
+optional = true
+
+[tool.poetry.group.optional_io_formats.dependencies]
+datapackage = ">=1.15.2"
+pandas-datareader = ">=0.10.0"
+unfccc_di_api = ">=3.0.1"
+xlrd = ">=2.0.1"
+
+[tool.poetry.group.optional_plotting]
+optional = true
+
+[tool.poetry.group.optional_plotting.dependencies]
+plotly = ">=5.19.0"
+
+[tool.poetry.group.tests]
+optional = true
+
+[tool.poetry.group.tests.dependencies]
+coverage = ">=6.5.0"
+coveralls = ">=3.3.1"
+pytest = ">=8.0.2"
+pytest-cov = ">=4.1.0"
+pytest-mpl = ">=0.17.0"
+
+[tool.poetry.group.tutorials]
+optional = true
+
+[tool.poetry.group.tutorials.dependencies]
+ipykernel = ">=6.29.3"
+jupyter_client = ">=8.6.0"
+nbconvert = ">= 7.3"
+nbformat = ">=5.9.2"
+pypandoc = ">=1.13"
+
+
+[build-system]
+requires = ["poetry-core"]
```

Comment:
Locally, at least, I see the following: 
```python
>>> import pyam
>>> pyam.__version__
'2.1.1.dev36+g6eb26fd'
```
and
```bash
(pyam-iamc-py3.12) fridolin@fridolin-Latitude-5520 ~/pyam (migrate/to-poetry)> (.venv) pip list | grep pyam
pyam-iamc                     2.1.0.post34.dev0+5c7db7b /home/fridolin/pyam
```
But I can specify the setup like you did, especially since ixmp4 is also doing it your way.

Reply:
If it's working that's ok from my side, no need to change anything.